modules/nw-ovn-kubernetes-migration-about-live.adoc

```diff
@@ -0,0 +1,166 @@
+
+* Cluster administrators must disable the following custom resources:
++
+** Egress Router
+** Egress IP
+** Egress Firewalls
+** Multicast
+
+== Checking Egress and multicast custom resources
+
+The following custom resources must be disabled before performing a live migration of OpenShift-SDN to OVN-Kubernetes:
+
+* Egress Router
+* Egress IP
+* Egress Firewalls
+* Multicast
+
+You can check the status of these custom resources (CRs) with the following procedure.
+
+.Procedure 
+
+. From the OpenShift Container Platform web console, navigate to *Observe* -> *Metrics*.
+
+. In the *Expression* box, add the following queries, and then click *Add query*:
++
+[source,promql]
++
+----
+network_attachment_definition_instances{networks="egress-router"} > 0
+----
++
+----
+sdn_controller_num_egress_ips > 0
+----
++
+----
+sdn_controller_num_egress_firewalls > 0
+----
++
+----
+sdn_controller_num_multicast_enabled_namespaces > 0
+----
+
+. Click *Run queries*.
++
+If any datapoints are found, you should not proceed with the OpenShift SDN to OVN-Kubernetes live migration procedure. Instead you should use the following information to disable these CRs:
+
+* link:https://docs.openshift.com/container-platform/4.14/networking/openshift_sdn/disabling-multicast.html[Disabling multicast for a project]
+* link:https://docs.openshift.com/container-platform/4.14/networking/openshift_sdn/removing-egress-firewall.html[Removing an egress firewall from a project]
+* link:https://access.redhat.com/solutions/4854561[Removing an egress IP address]
+* To remove an egress router, delete the pod. 
+
+= Using the live migration method to migrate from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin
+
+Administrators should ensure that all prerequisites have been met before beginning the migration process. Afterwards, the following procedure can be used to initiate the live migration from the OpenShift SDN network plugin to the OVN-Kubernetes network plugin.
+
+.Procedure
+
+. Enter the following command to add the `unsupported-red-hat-internal-testing` annotation to the cluster-level network configuration:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"unsupported-red-hat-internal-testing": "true"}}}'
+----
+
+. Enter the following command to patch the cluster-level networking configuration and initiative the migration from OpenShift SDN to OVN-Kubernetes:
++
+[source,terminal]
+----
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"metadata":{"annotations":{"network.openshift.io/network-type-migration":""}},"spec":{"networkType":"OVNKubernetes"}}'
+----
+
+After running these commands, the migration process begins. The time it takes to complete the process will vary depending on how many nodes are in your cluster. 
```

Comment:
We shall ask users to monitor the `status.conditions` of the `network.config` CR. Additionally, the annotation `network.openshift.io/network-type-migration` can be removed when the migration is completed.

Reply:
@pliurh Is this command correct? 

oc get network.config <network-config-name> -o=jsonpath='{.status.conditions}'

Reply:
cc @weliang1

Reply:
@stevsmit To verify the migration completed successfully through:

`$ oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed"
                "message": "Network type migration is completed",
$ oc get network.config.openshift.io cluster -o jsonpath='{.status.networkType}'
OVNKubernetes`

Reply:
This command can be used `oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .`

Reply:
@weliang1 @pliurh PTAL:

3. After running these commands, the migration process begins. The time it takes to complete the process will vary depending on how many nodes are in your cluster. Enter the following commands to check the status of the `network.config` custom resource:
~~~
$ oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed"
~~~
~~~
$ oc get network.config.openshift.io cluster -o jsonpath='{.status.networkType}'
~~~
~~~
$ oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .
~~~

Reply:
@stevsmit Here is the output from oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .

```
$ oc get network.config cluster -o=jsonpath='{.status.conditions}' | jq .
[
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationMTUReady"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationTargetCNIAvailable"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationTargetCNIInUse"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is not in progress",
    "reason": "NetworkTypeMigrationNotInProgress",
    "status": "Unknown",
    "type": "NetworkTypeMigrationOriginalCNIPurged"
  },
  {
    "lastTransitionTime": "2024-02-22T19:48:18Z",
    "message": "Network type migration is completed",
    "reason": "NetworkTypeMigrationCompleted",
    "status": "False",
    "type": "NetworkTypeMigrationInProgress"
  }
]
```

You may do not need `oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed" `any more.

Reply:
Removed $ oc get network.config.openshift.io cluster -o json | grep "Network type migration is completed"